src/libs/ReportActionsUtils.ts

```diff
@@ -635,6 +665,145 @@ function isNotifiableReportAction(reportAction: OnyxEntry<ReportAction>): boolea
+            {
+                kind: 'userMention',
+                content: mentions[0].content,
+                accountID: mentions[0].accountID,
+            },
+            {
+                kind: 'text',
+                content: ` ${Localize.translateLocal('common.and')} `,
+            },
+            {
+                kind: 'userMention',
+                content: lastMention.content,
+                accountID: lastMention.accountID,
+            },
+        );
+    } else {
+        mentions.forEach((mention) => {
+            messageElements.push(
+                {
+                    kind: 'userMention',
+                    accountID: mention.accountID,
+                    content: `${mention.content}`,
+                },
+                {
+                    kind: 'text',
+                    content: `, `,
+                },
+            );
+        });
+
+        messageElements.push(
+            {
+                kind: 'text',
+                content: `${Localize.translateLocal('common.and')} `,
+            },
+            {
+                kind: 'userMention',
+                content: lastMention.content,
+                accountID: lastMention.accountID,
+            },
+        );
+    }
+
+    const roomName = originalMessage?.roomName;
+    if (roomName) {
+        const preposition = isInviteAction ? ` ${Localize.translateLocal('workspace.invite.to')} ` : ` ${Localize.translateLocal('workspace.invite.from')} `;
+
+        if (originalMessage.reportID) {
+            messageElements.push(
+                {
+                    kind: 'text',
+                    content: preposition,
+                },
+                {
+                    kind: 'roomReference',
+                    roomName,
+                    roomID: originalMessage.reportID,
+                    content: roomName,
+                },
+            );
+        }
+    }
+
+    return messageElements;
+}
+
+function getMemberChangeMessageFragment(reportAction: OnyxEntry<ReportAction>): Message {
+    const messageElements: MemberChangeMessageElement[] = getMemberChangeMessageElements(reportAction);
+    const html = messageElements
+        .map((messageElement) => {
+            switch (messageElement.kind) {
+                case 'userMention':
+                    return `<mention-user accountID=${messageElement.accountID}></mention-user>`;
+                case 'roomReference':
+                    return `<a href="${environmentURL}/r/${messageElement.roomID}" target="_blank">${messageElement.roomName}</a>`;
+                default:
+                    return messageElement.content;
+            }
+        })
+        .join('');
+
+    return {
+        html: `<muted-text>${html}</muted-text>`,
+        text: reportAction?.message ? reportAction?.message[0].text : '',
+        type: CONST.REPORT.MESSAGE.TYPE.COMMENT,
+    };
+}
+
+/**
+ * Return room channel log message
+ *
+ * @param {Object} reportAction
```

Comment:
Do wee need JSDoc "typing" when we have proper TypeScript typing? I think it's not necessary in `.ts` files. I would drop it in the whole PR